pkg/kubernetes/filter.go

```diff
@@ -0,0 +1,49 @@
+package kubernetes
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+//go:generate counterfeiter . ManifestFilter
+type ManifestFilter interface {
+	FilterOnCluster([]unstructured.Unstructured, string) []unstructured.Unstructured
+	FilterWhereLabelDoesNotExist([]unstructured.Unstructured, string) []unstructured.Unstructured
+}
+
+type manifestFilter struct {
+	items []unstructured.Unstructured
+}
+
+func NewManifestFilter(items []unstructured.Unstructured) manifestFilter {
+	return manifestFilter{items: items}
+}
+
+func (f *manifestFilter) FilterOnCluster(cluster string) []unstructured.Unstructured {
+	filtered := []unstructured.Unstructured{}
+	for _, item := range f.items {
+		annotations := item.GetAnnotations()
+		if annotations != nil {
+			if strings.EqualFold(annotations[AnnotationSpinnakerMonikerCluster], cluster) {
+				filtered = append(filtered, item)
+			}
+		}
+	}
+
+	return filtered
+}
+
+func (f *manifestFilter) FilterWhereLabelDoesNotExist(label string) []unstructured.Unstructured {
```

Comment:
This one might be my fault, but shouldn't this be named `FilterOnLabel`? `FilterWhereLabelDoesNotExist` kind of implies that it will return a list where the label passed in does not exist, when this function is really grabbing a list where the label does exist.